internal/featuretests/v3/global_authorization_test.go

```diff
@@ -0,0 +1,710 @@
+						Services:               grpcCluster("extension/auth/extension"),
+						ClearRouteCache:        true,
+						IncludePeerCertificate: true,
+						StatusOnError: &envoy_type.HttpStatus{
+							Code: envoy_type.StatusCode_Forbidden,
+						},
+						TransportApiVersion: envoy_core_v3.ApiVersion_V3,
+					},
+				),
+				nil, "h2", "http/1.1"),
+		},
+		SocketOptions: envoy_v3.TCPKeepaliveSocketOptions(),
+	}
+
+	c.Request(listenerType).Equals(&envoy_discovery_v3.DiscoveryResponse{
+		TypeUrl: listenerType,
+		Resources: resources(t,
+			httpListener,
+			httpsListener,
+			statsListener()),
+	}).Status(p).IsValid()
+
+	c.Request(routeType).Equals(&envoy_discovery_v3.DiscoveryResponse{
+		TypeUrl: routeType,
+		Resources: resources(t,
+			envoy_v3.RouteConfiguration(
+				"https/foo.com",
+				envoy_v3.VirtualHost("foo.com",
+					&envoy_route_v3.Route{
+						Match:  routePrefix("/"),
+						Action: routeCluster("default/s1/80/da39a3ee5e"),
+						TypedPerFilterConfig: map[string]*anypb.Any{
+							"envoy.filters.http.ext_authz": protobuf.MustMarshalAny(
+								&envoy_config_filter_http_ext_authz_v3.ExtAuthzPerRoute{
+									Override: &envoy_config_filter_http_ext_authz_v3.ExtAuthzPerRoute_CheckSettings{
+										CheckSettings: &envoy_config_filter_http_ext_authz_v3.CheckSettings{
+											ContextExtensions: map[string]string{
+												"header_type": "proxy_config",
+												"header_1":    "message_1",
+												"header_2":    "message_2",
+											},
+										},
+									},
+								},
+							),
+						},
+					},
+				),
+			),
+			envoy_v3.RouteConfiguration(
+				"ingress_http",
+				envoy_v3.VirtualHost("foo.com",
+					&envoy_route_v3.Route{
+						Match:  routePrefix("/"),
+						Action: withRedirect(),
+					},
+				),
+			),
+		),
+	})
+}
+
+func globalExternalAuthorizationWithTLSAuthOverride(t *testing.T, rh cache.ResourceEventHandler, c *Contour) {
+	p := fixture.NewProxy("TLSProxy").
+		WithFQDN("foo.com").
+		WithSpec(contour_api_v1.HTTPProxySpec{
+			VirtualHost: &contour_api_v1.VirtualHost{
+				Fqdn: "foo.com",
+				TLS: &contour_api_v1.TLS{
+					SecretName: "certificate",
+				},
+				Authorization: &contour_api_v1.AuthorizationServer{
+					ExtensionServiceRef: contour_api_v1.ExtensionServiceReference{
+						Namespace: "auth",
+						Name:      "extension",
+					},
+					ResponseTimeout: defaultResponseTimeout.String(),
+					FailOpen:        true,
+					WithRequestBody: &contour_api_v1.AuthorizationServerBufferSettings{
+						MaxRequestBytes:     512,
+						PackAsBytes:         true,
+						AllowPartialMessage: true,
+					},
+				},
+			},
+			Routes: []contour_api_v1.Route{
+				{
+					Services: []contour_api_v1.Service{
+						{
+							Name: "s1",
+							Port: 80,
+						},
+					},
+				},
+			},
+		})
+
+	rh.OnAdd(p)
+
+	var httpListener = defaultHTTPListener()
+
+	// replace the default filter chains with an HCM that includes the global
```

Comment:
nit/refactor: could pull this common block setting up the HCM into a helper

Reply:
moved this into a common function